### ISSUE TITLE
GLMakie window wont close after Ctrl-C in the REPL

### ISSUE THREAD

**Issue by rafaqz** [MakieOrg/Makie.jl#2461]
After using Ctrl-C in the REPL when a GLMakie window is running, the window can no longer be closed. The only option seems to be closing it with `xkill`. But this leaves open GL unavailable for reopening a new window.

This is on Arch Linux, GLMakie v0.7.3.

**Comment by rafaqz** [comment]
To clarify this happens when I have a process waiting on Makie.jl to finish, and I Control-C out of that process.

**Comment by rafaqz** [comment]
```julia
using GLMakie
function f()
    fig = Figure()
    screen = display(fig)
    while screen.window_open[]
        sleep(0.1)
    end
end

f()
```

**Comment by felixcremer** [comment]
I can reproduce this on Ubuntu and after I killed the function, the Makie window stays open and it is force quited by the operating system and I get the following stacktrace:

```julia
julia> f()
^C┌ Warning: error in renderloop
│   exception =
│    InterruptException:
│    Stacktrace:
│     [1] poptask(W::Base.InvasiveLinkedListSynchronized{Task})
│       @ Base ./task.jl:921
│     [2] wait()
│       @ Base ./task.jl:930
│     [3] wait(c::Base.GenericCondition{Base.Threads.SpinLock})
│       @ Base ./condition.jl:124
│     [4] _trywait(t::Timer)
│       @ Base ./asyncevent.jl:138
│     [5] wait
│       @ ./asyncevent.jl:155 [inlined]
│     [6] sleep(sec::Float64)
│       @ Base ./asyncevent.jl:240
│     [7] on_demand_renderloop(screen::GLMakie.Screen{GLFW.Window})
│       @ GLMakie ~/.julia/packages/GLMakie/6Blp9/src/screen.jl:844
│     [8] renderloop(screen::GLMakie.Screen{GLFW.Window})
│       @ GLMakie ~/.julia/packages/GLMakie/6Blp9/src/screen.jl:858
│     [9] (::GLMakie.var"#65#66"{GLMakie.Screen{GLFW.Window}})()
│       @ GLMakie ./task.jl:484
└ @ GLMakie ~/.julia/packages/GLMakie/6Blp9/src/screen.jl:867
ERROR: InterruptException:
Stacktrace:
 [1] try_yieldto(undo::typeof(Base.ensure_rescheduled))
   @ Base ./task.jl:871
 [2] wait()
   @ Base ./task.jl:931
 [3] wait(c::Base.GenericCondition{Base.Threads.SpinLock})
   @ Base ./condition.jl:124
 [4] _trywait(t::Timer)
   @ Base ./asyncevent.jl:138
 [5] wait
   @ ./asyncevent.jl:155 [inlined]
 [6] sleep(sec::Float64)
   @ Base ./asyncevent.jl:240
 [7] f()
   @ Main ./REPL[3]:5
 [8] top-level scope
   @ REPL[4]:1

julia> Killed
```

**Comment by rafaqz** [comment]
That fixed it! So call that instead of using xkill. Is there a way that GLMakie can know to call that if it has an error trying to make a new window?

**Comment by SimonDanisch** [comment]
Hm not sure... The main reason that windows are freezing is because we do `GLFW.PollEvents()` in the renderloop, and if that crashes/gets interrupted or whatever, we stop polling, which will make the OS think the window froze.
Maybe the easiest thing to improve the situation would be to move the polling to its own task.

**Comment by rafaqz** [comment]
Ok probably not high priority either

**Comment by lassepe** [comment]
Carrying over the suggestion from https://github.com/MakieOrg/Makie.jl/issues/3114#issuecomment-2119233122

This
```julia
GLMakie.SINGLETON_SCREEN[1].rendertask = nothing
```
allows Makie to recover. Maybe we can automate that.
